### ISSUE TITLE
Nodes marked as `isLink` are tested against `MinimumTapTargetGuideline`

### ISSUE THREAD

**Issue by NickAb** [flutter/flutter#93867]
`MinimumTapTargetGuideline` includes checking elements marked as links (`SemanticsFlag.isLink`) and fails if they don't meet target size.

Platforms define suggested touch target size for interactive elements, but inline links are different from standalone interactive elements.  
Text with links (i.e. RichView with TextSpan that has gesture recogniser) can is very similar to a web view and (probably) WCAG recommendations should be followed for inline links.

According to [WCAG](https://www.w3.org/WAI/WCAG21/Understanding/target-size.html):
> Links within a paragraph of text have varying touch target dimensions. Links within paragraphs of text do no need to meet the 44 by 44 CSS pixels requirements.
> A text link that is in a sentence is excluded and does not need to meet the 44 by 44 CSS pixel requirement. If the text link is the full sentence, then the text link target touch area does need to meet the 44 by 44 CSS pixels. 

I.e. nodes marked as `isLink` should be skipped by `MinimumTapTargetGuideline`.

**Steps to Reproduce**

1. Apply https://github.com/flutter/flutter/commit/4114dad30d460d9a9417c1aedb3a4c1a93259298.patch
2. Run tests
    ```
    cd packages/flutter_test
    flutter test test/accessibility_test.dart
    ```
**Expected results:**

- All tests should pass.

**Actual results:**

- The following tests fail:
  - `tap target size guideline` `Does not fail on links`

**Comment by NickAb** [comment]
I don't have permission to edit labels, correct labels should be: `a: accessibility` and `a: tests`, not `team: infra`. Sorry about that.
